src/plugins/data/server/index_patterns/register_index_pattern_usage_collection.ts

```diff
@@ -0,0 +1,193 @@
+      scriptedFieldLineCount: { ...countSummaryDefaults },
+      runtimeFieldLineCount: { ...countSummaryDefaults },
+    },
+  };
+
+  await ids.reduce(async (col, id) => {
+    await col;
+    const ip = await indexPatterns.get(id);
+
+    const scriptedFields = ip.getScriptedFields();
+    const runtimeFields = ip.fields.filter((fld) => !!fld.runtimeField);
+
+    if (scriptedFields.length > 0) {
+      // increment counts
+      results.indexPatternsWithScriptedFieldCount++;
+      results.scriptedFieldCount += scriptedFields.length;
+
+      // calc LoC
+      results.perIndexPattern.scriptedFieldLineCount = minMaxAvgLoC(
+        scriptedFields.map((fld) => fld.script || '')
+      );
+
+      // calc field counts
+      results.perIndexPattern.scriptedFieldCount.min = updateMin(
+        results.perIndexPattern.scriptedFieldCount.min,
+        scriptedFields.length
+      );
+      results.perIndexPattern.scriptedFieldCount.max = updateMax(
+        results.perIndexPattern.scriptedFieldCount.max,
+        scriptedFields.length
+      );
+      results.perIndexPattern.scriptedFieldCount.avg =
+        results.scriptedFieldCount / results.indexPatternsWithScriptedFieldCount;
+    }
+
+    if (runtimeFields.length > 0) {
+      // increment counts
+      results.indexPatternsWithRuntimeFieldCount++;
+      results.runtimeFieldCount += runtimeFields.length;
+
+      // calc LoC
+      const runtimeFieldScripts = runtimeFields.map(
+        (fld) => fld.runtimeField?.script?.source || ''
+      );
+      results.perIndexPattern.runtimeFieldLineCount = minMaxAvgLoC(runtimeFieldScripts);
+
+      // calc field counts
+      results.perIndexPattern.runtimeFieldCount.min = updateMin(
+        results.perIndexPattern.runtimeFieldCount.min,
+        runtimeFields.length
+      );
+      results.perIndexPattern.runtimeFieldCount.max = updateMax(
+        results.perIndexPattern.runtimeFieldCount.max,
+        runtimeFields.length
+      );
+      results.perIndexPattern.runtimeFieldCount.avg =
+        results.runtimeFieldCount / results.indexPatternsWithRuntimeFieldCount;
+    }
+  }, Promise.resolve());
+
+  return results;
+}
+
+export function registerIndexPatternsUsageCollector(
+  getStartServices: StartServicesAccessor<DataPluginStartDependencies, DataPluginStart>,
+  usageCollection?: UsageCollectionSetup
+): void {
+  if (!usageCollection) {
+    return;
+  }
+
+  const indexPatternUsageCollector = usageCollection.makeUsageCollector<IndexPatternUsage>({
+    type: 'index-patterns',
+    isReady: () => true,
+    fetch: async () => {
+      const [{ savedObjects, elasticsearch }, , { indexPatterns }] = await getStartServices();
+      const indexPatternService = await indexPatterns.indexPatternsServiceFactory(
+        new SavedObjectsClient(savedObjects.createInternalRepository()),
+        elasticsearch.client.asInternalUser
+      );
+
+      return await getIndexPatternTelemetry(indexPatternService);
+    },
+    schema: {
```

Comment:
We highly recommend adding descriptions to new usage collector schemas so that you don't need to come back in 6 months and do it then. The descriptions will ultimately be used to build up an elastic telemetry data dictionary.
Descriptions are optional at the moment but will eventually be mandatory.